Sources/Datadog/RUM/Integrations/RUMContextAttributes.swift

```diff
@@ -10,11 +10,11 @@ import Foundation
 internal enum RUMContextAttributes {
     internal enum IDs {
         /// The ID of RUM application (`String`).
-        internal static let applicationID = "application_id"
+        internal static let applicationID = "application.id"
 
         /// The ID of current RUM session (standard UUID `String`, lowercased).
         /// In case the session is rejected (not sampled), RUM context is set to empty (`[:]`) in core.
-        internal static let sessionID = "session_id"
+        internal static let sessionID = "session.id"
```

Comment:
blocker 🚫/ This change break Session Replay, which expects `"application_id"` and `"session_id"`. With being unable to recognise these new values SR thinks there is no RUM, so it doesn't record anything.

1. We need to update these values [here](https://github.com/DataDog/dd-sdk-ios/blob/10135195869d7f785f7a63201e4ecd84795b0490/DatadogSessionReplay/Sources/DatadogCoreIntegration/RUMContextReceiver.swift#L14-L16).
2. Also please update these values in comments of RUM dependency definition within SR module [here](https://github.com/DataDog/dd-sdk-ios/blob/ncreated/RUMM-3320/GH-1292-fix-apm-with-rum/DatadogSessionReplay/Sources/DatadogCoreIntegration/RUMDependency.swift) - this definition is a central place where SR defines everything it expects or provides to other modules in SDK context.

I bump the need for 2 things:
- Integration Tests for SR: it is now possible on `feature/v2` - I created [REPLAY-1863](https://datadoghq.atlassian.net/browse/REPLAY-1863)
- Integration Unit Tests: to validate feature's contract in unit tests - @maxep do we have JIRA for this?

Reply:
> Integration Unit Tests: to validate feature's contract in unit tests - @maxep do we have JIRA for this?